bdsf/islands.py

```diff
@@ -100,6 +105,26 @@ def __call__(self, img):
             img.nisl = len(img.islands)
             img.pyrank = det_img.pyrank
             img.minpix_isl = det_img.minpix_isl
+
+            if opts.output_all:
+                write_islands(img)
+            if opts.savefits_rankim or opts.output_all:
+                func.write_image_to_file(img.use_io, img.imagename + '_pyrank.fits', img.pyrank, img)
+            if opts.savefits_det_rmsim or opts.output_all:
+                resdir = img.basedir + '/background/'
+                if not os.path.exists(resdir):
+                    os.makedirs(resdir)
+                func.write_image_to_file(img.use_io, img.imagename + '.detection_rmsd_I.fits',
+                                         img.detection_rms_arr, img, resdir)
+                mylog.info('%s %s' % ('Writing ', resdir+img.imagename+'.detection_rmsd_I.fits'))
+            if opts.savefits_det_meanim or opts.output_all:
+                resdir = img.basedir + '/background/'
+                if not os.path.exists(resdir):
+                    os.makedirs(resdir)
```

Comment:
And here you could use the `exist_ok=True`, so you don't have to check for existence first.